fishy/engine/fullautofisher/engine.py

```diff
@@ -136,39 +140,46 @@ def get_coords(self):
         return get_values_from_image(img)
 
     def move_to(self, target) -> bool:
-        cancel_states = [fishing_mode.State.FIGHT, fishing_mode.State.REELIN,
-                        fishing_mode.State.FISHING]
-        if FishingMode.CurrentMode not in cancel_states:
-            current = self.get_coords()
-            if not current:
-                return False
+        logging.info("engine.move_to()")
+        current = self.get_coords()
+        if not current:
+            return False
 
-            print(f"Moving from {(current[0], current[1])} to {target}")
-            move_vec = target[0] - current[0], target[1] - current[1]
+        print(f"Moving from {(current[0], current[1])} to {target}")
+        move_vec = target[0] - current[0], target[1] - current[1]
 
-            dist = math.sqrt(move_vec[0] ** 2 + move_vec[1] ** 2)
-            print(f"distance: {dist}")
-            if dist < 5e-05:
-                print("distance very small skipping")
-                return True
+        dist = math.sqrt(move_vec[0] ** 2 + move_vec[1] ** 2)
+        print(f"distance: {dist}")
+        if dist < 5e-05 and self.scan_counter <= 4:
+        # if dist < 5e-05:
+            print("distance very small skipped {} move".format(self.scan_counter + 1))
+            self.scan_counter += 1
+            return True
+        else:
+            logging.info("move_to() not skipped")
+            self.scan_counter = 0
+            pass
 
-            target_angle = math.degrees(math.atan2(-move_vec[1], move_vec[0])) + 90
-            from_angle = current[2]
+        target_angle = math.degrees(math.atan2(-move_vec[1], move_vec[0])) + 90
+        from_angle = current[2]
 
-            if not self.rotate_to(target_angle, from_angle):
-                return False
+        if not self.rotate_to(target_angle, from_angle):
+            return False
 
-            walking_time = dist / self.calibrator.move_factor
+        walking_time = dist / self.calibrator.move_factor
+        while FishingMode.CurrentMode not in cancel_states:
             print(f"walking for {walking_time}")
             kb.press(fishing_event.FishEvent.walk_key)
             time.sleep(walking_time)
             kb.release(fishing_event.FishEvent.walk_key)
             print("done")
             return True
         else:
-            return False
+            pass
```

Comment:
looks like  you have inverted the if statement, ignore the comment on old commit